Lightyear/Core.idr

```diff
@@ -101,15 +152,15 @@ infixl 3 <|>|
 (<|>|) : Monad m => ParserT str m a
                  -> Lazy (ParserT str m a)
                  -> ParserT str m a
-(<|>|) (PT x) y = PT $ \r, us, cs, ue, ce, i =>
-  x r us cs (\err => let PT y' = y in y' r us cs (ue . (err ++))
-                                                 (ce . (err ++)) i) ce i
+(<|>|) = lightyearAlt'
```

Comment:
I'm not sure if this will be sufficiently lazy; have you tested it, by any chance?